src/scrapy_settings_log/__init__.py

```diff
@@ -51,6 +51,16 @@ def spider_closed(self, spider):
         regex = settings.get("SETTINGS_LOGGING_REGEX")
         if regex is not None:
             settings = {k: v for k, v in settings.items() if re.search(regex, k)}
+        if settings.get("MASKED_SENSITIVE_SETTINGS_ENABLED", True):
+            default_regexes = [
+                ".*(?i)(api[\W_]*key).*",  # apikey and possible variations e.g: shub_apikey or SC_APIKEY
+                ".*(?i)(AWS[\W_]*SECRET[\W_]*ACCESS[\W_]*KEY).*",  # AWS_SECRET_ACCESS_KEY and possible variations
+                ".*(?i)([\W_]*password[\W_]*).*"  # password word
+            ]
+            regex_list = settings.get("MASKED_SENSITIVE_SETTINGS_REGEX_LIST", default_regexes)
+            for reg in regex_list:
+                updated_settings = {k: '*'*len(v) if v else v for k, v in settings.items() if re.match(reg, k)}
```

Comment:
**thought (non-blocking):** We might want to return a fixed-length string with "*"s for extra security (so the password length isn't leaked) but it might not be too problematic either way.